test/project_types/theme/commands/generate/env_test.rb

```diff
@@ -0,0 +1,189 @@
+              .returns(THEMES)
+            CLI::UI::Prompt.expects(:ask)
+              .with(context.message('theme.generate.env.ask_theme'))
+              .returns('2468')
+
+            Themekit.expects(:generate_env)
+              .with(context,
+                    password: 'boop',
+                    themeid: '2468',
+                    store: 'shop.myshopify.com',
+                    env: nil)
+              .returns(true)
+
+            command = Theme::Commands::Generate::Env.new(context)
+            command.call
+          end
+        end
+
+        def test_can_use_default_shop_and_password
+          FakeFS do
+            File.write('config.yml', CONFIG.to_yaml)
+            context = ShopifyCli::Context.new
+
+            CLI::UI::Prompt.expects(:ask)
+              .with(context.message('theme.generate.env.ask_store_default', 'shop.myshopify.com'))
+              .returns('')
+
+            CLI::UI::Prompt.expects(:ask)
+              .with(context.message('theme.generate.env.ask_password_default', 'boop'))
+              .returns('')
+
+            Themekit.expects(:query_themes)
+              .with(context, store: 'shop.myshopify.com', password: 'boop')
+              .returns(THEMES)
+            CLI::UI::Prompt.expects(:ask)
+              .with(context.message('theme.generate.env.ask_theme'))
+              .returns('2468')
+
+            Themekit.expects(:generate_env)
+              .with(context,
+                    password: 'boop',
+                    themeid: '2468',
+                    store: 'shop.myshopify.com',
+                    env: nil)
+              .returns(true)
+
+            command = Theme::Commands::Generate::Env.new(context)
+            command.call
+          end
+        end
+
+        def test_can_provide_credentials_with_flags
+          FakeFS do
+            File.write('config.yml', CONFIG.to_yaml)
+            context = ShopifyCli::Context.new
+
+            CLI::UI::Prompt.expects(:ask)
+              .with(context.message('theme.generate.env.ask_store_default', 'shop.myshopify.com'))
+              .never
+
+            CLI::UI::Prompt.expects(:ask)
+              .with(context.message('theme.generate.env.ask_password_default', 'boop'))
+              .never
+
+            Themekit.expects(:query_themes)
+              .with(context, store: 'shop.myshopify.com', password: 'boop')
+              .never
+            CLI::UI::Prompt.expects(:ask)
+              .with(context.message('theme.generate.env.ask_theme'))
+              .never
+
+            Themekit.expects(:generate_env)
+              .with(context,
+                    password: 'boop',
+                    themeid: '2468',
+                    store: 'shop.myshopify.com',
+                    env: nil)
+              .returns(true)
+
+            command = Theme::Commands::Generate::Env.new(context)
+            command.options.flags[:store] = 'shop.myshopify.com'
```

Comment:
Same thing here, we probably want to use different values to test that the defaults are overridden.